### PR TITLE
PositionKit as QML singleton

### PR DESCRIPTION
**Pull request by tomasMizera**
Changed position kit to be a singleton in QML. Previously we needed to pass `PositionKit` qml instance all over the place. It is handy to be able to access device position from any part of the app.

### CODE REVIEW

File: app/qml/misc/GpsDataPage.qml [hunk 1]
```diff
@@ -35,6 +34,13 @@ Page {
     }
   }
 
+  MapPosition {
+    id: mapPositioning
+
+    positionKit: __positionKit
```
Comment (PeterPetrik) [inline app/qml/misc/GpsDataPage.qml:40]: maybe we can use singleton and not pass it around?

File: app/loader.cpp [hunk 2]
```diff
@@ -308,18 +302,6 @@ void Loader::appStateChanged( Qt::ApplicationState state )
 
   logHelper << "Application changed state to: " << state;
   CoreUtils::log( "Input", msg );
-
```
Comment (PeterPetrik) [inline app/loader.cpp:311]: should here be called PositionKit::appStateChanged?

Comment (tomasMizera) [reply app/loader.cpp:311]: Well, I connected it directly from main.cpp [here](https://github.com/lutraconsulting/input/pull/1888/files#diff-c1b709b8a90bef4573ddad8a9b2fbc5d5dbc85e4dc759f95b6d9905931fec24fR450) in order to remove the coupling of loader (which from my point of view should serve only to load a project) and position.

File: app/qml/misc/GpsDataPage.qml (hunk 1 above)
Comment (tomasMizera) [reply app/qml/misc/GpsDataPage.qml:40]: You mean to convert `MapPosition` to singleton too?

Comment (tomasMizera) [reply app/qml/misc/GpsDataPage.qml:40]: `MapPosition` is a cpp class that needs mapSettings to work. MapSettings is not a singleton class so we would need to pass new map settings to it each time project is loaded

Comment (PeterPetrik) [reply app/qml/misc/GpsDataPage.qml:40]: No I mean why you need to have `positionKit: __positionKit` here and not use `__positionKit` directly in the element?

File: app/loader.cpp (hunk 2 above)
Comment (PeterPetrik) [reply app/loader.cpp:311]: ok fair enough. is the original "Loader: appStateChanged" needed at all now?

**Review by PeterPetrik — APPROVED** [review]
good to go

File: app/qml/misc/GpsDataPage.qml (hunk 1 above)
Comment (tomasMizera) [reply app/qml/misc/GpsDataPage.qml:40]: Because it is a C++ class.

**Comment by tomasMizera** [comment]
@PeterPetrik there was actually one place where I was still passing the positionKit :D `PositionMarker.qml`

File: app/qml/misc/GpsDataPage.qml (hunk 1 above)
Comment (PeterPetrik) [reply app/qml/misc/GpsDataPage.qml:40]: make sense , sorry :)

**Comment by tomasMizera** [comment]
@PeterPetrik Autotests (purchasing tests) are failing due to a known old issue with test db.

**Pull request merged**